pkg/router/controller/controller.go

```diff
@@ -111,6 +112,12 @@ func (c *RouterController) HandleEndpoints() {
 		return
 	}
 
+	// NB: svc might be nil here
+	svc, err := c.LookupService(endpoints)
```

Comment:
I don't like this - I think the lookup for service should be done by the specific handler, not by passing this down.  When we switch to informer patterns in the router, there will be no way to pass this down, so let's not start now.